source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/DefaultChargeLinkSettingRepositoryTests.cs

```diff
@@ -0,0 +1,116 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using GreenEnergyHub.Charges.Domain.Charges;
+using GreenEnergyHub.Charges.Domain.MeteringPoints;
+using GreenEnergyHub.Charges.Infrastructure.Context;
+using GreenEnergyHub.Charges.Infrastructure.Context.Model;
+using GreenEnergyHub.Charges.Infrastructure.Repositories;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+using Xunit.Categories;
+using Charge = GreenEnergyHub.Charges.Infrastructure.Context.Model.Charge;
+using ChargeOperation = GreenEnergyHub.Charges.Infrastructure.Context.Model.ChargeOperation;
+using DBDefaultChargeLinkSetting = GreenEnergyHub.Charges.Infrastructure.Context.Model.DefaultChargeLinkSetting;
+using MarketParticipant = GreenEnergyHub.Charges.Infrastructure.Context.Model.MarketParticipant;
+
+namespace GreenEnergyHub.Charges.Tests.Infrastructure.Repositories
+{
+    /// <summary>
+    /// Tests <see cref="DefaultChargeLinkRepository"/> using an SQLite in-memory database.
+    /// </summary>
+    [UnitTest]
+    public class DefaultChargeLinkSettingRepositoryTests
+    {
+        private readonly DbContextOptions<ChargesDatabaseContext> _dbContextOptions =
+            new DbContextOptionsBuilder<ChargesDatabaseContext>()
+                .UseSqlite("Filename=Test.db")
+                .Options;
+
+        private int _expectedChargeRowId;
+
+        [Fact]
+        public async Task GetDefaultChargeLinks_WhenCalledWithMeteringPointType_ReturnsDefaultCharges()
+        {
+            // Arrange
+            CreateAndSeedDatabase();
+            await using var chargesDatabaseContext = new ChargesDatabaseContext(_dbContextOptions);
+            var sut = new DefaultChargeLinkRepository(chargesDatabaseContext);
+
+            // Act
+            var actual = await
+                sut.GetDefaultChargeLinksAsync(MeteringPointType.Consumption).ConfigureAwait(false);
+
+            // Assert
+            var actualDefaultChargeLinkSettings = actual as DefaultChargeLink[] ?? actual.ToArray();
+
+            actualDefaultChargeLinkSettings.Should().NotBeNullOrEmpty();
+            actualDefaultChargeLinkSettings.First().ChargeRowId.Should().Be(_expectedChargeRowId);
+        }
+
+        private void CreateAndSeedDatabase()
+        {
+            using var context = new ChargesDatabaseContext(_dbContextOptions);
+            context.Database.EnsureDeleted();
+            context.Database.EnsureCreated();
+            var marketParticipant = context.MarketParticipants.Add(
+                new MarketParticipant { Name = "Name", Role = 1, MarketParticipantId = "MarketParticipantId" });
+            context.SaveChanges();
+            var charge = context.Charges.Add(new Charge
+            {
+                Currency = "DKK",
+                Resolution = 1,
```

Comment:
I have no clue of any of all these integer values. How about replacering all like this:

```suggestion
                Resolution = (int)Resolution.PT15M,
```